apm-sniffer/apm-sdk-plugin/dubbo-plugin/src/main/java/org/apache/skywalking/apm/plugin/dubbo/DubboInterceptor.java

```diff
@@ -56,6 +57,9 @@ public void beforeMethod(EnhancedInstance objInst, Method method, Object[] allAr
         Class<?>[] argumentsTypes, MethodInterceptResult result) throws Throwable {
         Invoker invoker = (Invoker)allArguments[0];
         Invocation invocation = (Invocation)allArguments[1];
+        if (invocation.getAttachments().containsKey(SW3CarrierItem.HEADER_NAME)) {
+            invocation.getAttachments().remove(SW3CarrierItem.HEADER_NAME);
+        }
```

Comment:
Remove should be something similar than this

```
            CarrierItem next = contextCarrier.items();
            while (next.hasNext()) {
                next = next.next();
                rpcContext.getAttachments().put(next.getHeadKey(), next.getHeadValue());
            }
```

Key is defined in there.

Reply:
When dubbo retry, the parameter sw3 will not be updated.

Reply:
Therefore, we should clear sw3 in the invocation first,  so that the latest sw3 will be placed according to the RpcContext

Reply:
> When dubbo retry, the parameter sw3 will not be updated.

@tzy1316106836 what @wu-sheng meant is that you should not simply use `SW3CarrierItem.HEADER_NAME` as key to remove the context, you should use something like：

```java
            CarrierItem next = contextCarrier.items();
            while (next.hasNext()) {
                next = next.next();
                rpcContext.getAttachments().remove(next.getHeadKey());
            }
```

because there are chances that the users are using SkyWalking 6.x, which uses protocol v2 and the key is `SW6CarrierItem.HEADER_NAME` instead of `SW3CarrierItem.HEADER_NAME`, what's more, `SW3CarrierItem` will be removed soon

Reply:
done